edb/server/protocol/auth_ext/__init__.py

```diff
@@ -0,0 +1,86 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2022-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import http
+import json
+import urllib.parse
+
+from edb import errors
+from edb.common import debug
+from edb.common import markup
+
+from . import oauth
+
+
+async def handle_request(
+    request, response, db, extension_base_path: str, args: list[str]
+):
+    try:
+        # Set up routing to the appropriate handler
+        if args[0] == "authorize":
+            provider_name: str | None = _get_search_param(
```

Comment:
(not something to fix) repeating the type here feels a bit redundant to me as `_get_search_param()` already defined a returning type 🤔

Reply:
Yeah, good call, I personally prefer to lean into inference when I can. I think I had this in place before I factored out the `_get_search_param` function. Will fix!